test/uk/gov/hmrc/agentaccesscontrol/controllers/AuthorisationControllerSpec.scala

```diff
@@ -109,12 +114,35 @@ class AuthorisationControllerSpec extends UnitSpec with BeforeAndAfterEach with
 
       whenMtdAuthorisationServiceIsCalled thenReturn(Future failed new IllegalStateException("some error"))
 
-      an[IllegalStateException] shouldBe thrownBy(status(controller.isAuthorisedForMtdSa(AgentCode(""), MtdClientId("utr"))(fakeRequest)))
+      an[IllegalStateException] shouldBe thrownBy(status(controller().isAuthorisedForMtdSa(AgentCode(""), MtdClientId("utr"))(fakeRequest)))
+    }
+
+    "isAuthorisedForPaye" should {
+
+      "return 200 when Paye is enabled" in {
+        whenPayeAuthorisationServiceIsCalled thenReturn(Future successful true)
+
+        val response = controller().isAuthorisedForPaye(AgentCode(""), EmpRef("123", "123456"))(fakeRequest)
+
+        status(response) shouldBe 200
+      }
+
+      "return 403 when Paye is disabled" in {
+        whenPayeAuthorisationServiceIsCalled thenReturn(Future successful true)
+        val payeEnabled = false
+
+        val response = controller(payeEnabled).isAuthorisedForPaye(AgentCode(""), EmpRef("123", "123456"))(fakeRequest)
```

Comment:
it might be neater to remove the `val payeEnabled = false` above and then do `controller(payeEnabled = true)` here

you can leave it as is though if you prefer